presto-main/src/main/java/io/prestosql/operator/window/LeadFunction.java

```diff
@@ -51,8 +54,19 @@ public void processRow(BlockBuilder output, int frameStart, int frameEnd, int cu
             checkCondition(offset >= 0, INVALID_FUNCTION_ARGUMENT, "Offset must be at least 0");
 
             long valuePosition = currentPosition + offset;
+            boolean withinParition = (valuePosition >= 0) && (valuePosition < windowIndex.size());
 
-            if ((valuePosition >= 0) && (valuePosition < windowIndex.size())) {
+            if (ignoreNulls) {
+                while (withinParition) {
+                    if (!windowIndex.isNull(valueChannel, toIntExact(valuePosition))) {
+                        break;
+                    }
+                    valuePosition++;
+                    withinParition = (valuePosition >= 0) && (valuePosition < windowIndex.size());
+                }
+            }
+
```

Comment:
I am not sure about the semantics of this implementation. You start at `currentPosition + offset`, like it was in the previous implementation which respected nulls, and then look forward for the first non-null value. However, from reading the spec, it appears to me that you should instead start at `currentPosition` and go forward unless you encounter `offset`-th non-null value.
The same applies to `LagFunction` and `NthValueFunction`.